Makefile

```diff
@@ -1374,11 +1374,54 @@ mpi_f08_test:
 	$(if $(findstring 1,$(MPAS_MPI_F08)), $(eval MPI_F08_MESSAGE = "Using the mpi_f08 module."), )
 	$(if $(findstring 1,$(MPAS_MPI_F08)), $(info mpi_f08 module detected.))
 
+
+pnetcdf_test:
+	@#
+	@# Create test C programs that look for PNetCDF header file and some symbols in it
+	@#
+ifneq "$(PNETCDF)" ""
+	@echo "Checking if a test PNetCDF C program will build..."
+	@printf "#include <pnetcdf.h>\n\
+			&#include <mpi.h>\n\
+			&int main(){\n\
+			&    MPI_Info info;\n\
+			&    int err=0, cmode=NC_NOCLOBBER, ncid;\n\
+			&    err = ncmpi_create(MPI_COMM_WORLD, \"foo.nc\", cmode, info, &ncid);\n\
+			&    return 0;\n\
+			&}\n" | sed 's/&/ /' > pnetcdf.c
+	@( $(CC) pnetcdf.c $(CPPINCLUDES) $(CFLAGS) $(LDFLAGS) -L$(PNETCDF)/$(PNETCDFLIBLOC) -lpnetcdf  -o pnetcdf.out > pnetcdf.log 2>&1; \
```

Comment:
Note that `LIBS` isn't used here, instead only the parts related to the PNetCDF library. This is because the smiol and smiolf libraries haven't been built yet.

This may not be the best way to do this. Suggestions are welcome!

Reply:
This seems reasonable to me!